sklearn/impute.py

```diff
@@ -1207,6 +1208,18 @@ def fit(self, X, y=None):
             raise ValueError("'features' has to be either 'missing-only' or "
                              "'all'. Got {} instead.".format(self.features))
 
+        if self.sparse is True and self.missing_values == 0:
```

Comment:
I don't think the sparsity of the *output* is the problem, only that of the *input*.

Reply:
Ok, so in this case we will pass a sparse array without densifying it with missing_value=0 (which is implicit in sparse array)?

Reply:
The output is only ones and zeros. It has nothing to do with the value of `missing_values`. We only want to raise an error when the input is a sparse matrix and missing values == 0